driver/src/main/scala/api/commands/writeresult.scala

```diff
@@ -32,8 +32,7 @@ sealed trait WriteResult {
 object WriteResult {
   def lastError(result: WriteResult): Option[LastError] = result match {
```

Comment:
Considering `lastError` this change is ok.

Considering the bulk use case mentioned, it should be the only case where a write op is `ok` but with `writeErrors`.

Additionally, I don't think this change is sufficient to have all the bulk errors in the corresponding `MultiBulkWriteResult`.